backend/src/models/db/tables.js

```diff
@@ -99,121 +134,96 @@ function createReviewTable (db) {
+    const query = `INSERT INTO subject (${columns}) VALUES (${placeholders})`
+    const prep = db.prepare(query)
+
+    // Do insertions and return promise for all of them to be completed
+    const promises = subjectData.map(subj => insertDB(db, 'subject', subj, prep))
+    return Promise.all(promises)
+}
+
+function initCourseTable(db) {
+    // Kind of hacky, but we only serving one university at this time
+    const universityID = 1
+    courseData.forEach((subj) => { subj['universityID'] = universityID })
+
+    // Prepare query
+    const columns = Object.keys(courseData[0])
+    const placeholders = columns.map(_ => '?').join()
+    const query = `INSERT INTO course (${columns}) VALUES (${placeholders})`
+    const prep = db.prepare(query)
+
+    // Do insertions and return promise for all of them to be completed
+    const promises = courseData.map(subj => insertDB(db, 'course', subj, prep))
+    return Promise.all(promises)
 }
 
 /**
- * Initiates a new SQL database from the given param
- * @param   {string} databaseName A db name, if not :memory: initiates a .db file
+ * Initiates a new SQL database by creating the tables with some UNSW data
+ * @param   {object} SQLObject
  * @returns {object} SQLObject
  */
-function createDB(databaseName) {
-    // Create the database object
-    const db = new sqlite3.Database(databaseName)
-
-    // Create the database tables
+function createDB(db) {
     db.serialize(() => {
-        createUserTable(db)
-        createUniversityTable(db)
-        createSubjectTable(db)
-        createCourseTable(db)
-        createQuestionTable(db)
-        createReviewTable(db)
-        createCommentTable(db)
+        db = createUserTable(db)
+        db = createUniversityTable(db)
+        db = createSubjectTable(db)
+        db = createCourseTable(db)
+        db = createQuestionTable(db)
+        db = createReviewTable(db)
+        db = createCommentTable(db)
     })
-
-    return db
+    return Promise.all([initUniTable(db), initSubjectTable(db), initCourseTable(db)])
 }
 
 // Insert given JSON object into database table.
 // data = { column : value }
 // For security reasons, column inputs can NEVER be user defined.
-// TODO - FIX ISSUE WHERE this.lastID != user/uni/course/question/answerID
-//        THIS IS DUE TO THE FACT THAT WE CAN ONLY RETURN THE ROW_ID AFTER
-//        INSERTION WITHOUT A LOOKUP... TOO TIRED TO SOLVE RN
-function insertDB (db, table, data) {
+function insertDB (db, table, data, prep) {
     return new Promise((resolve, reject) => {
-        const columns = Object.keys(data)
-        const placeholders = columns.map(_ => '?').join()
-        const query = `INSERT INTO ${table} (${columns}) VALUES (${placeholders})`
-        db.run(
-            query,
-            Object.values(data),
-            function (err) { err ? reject(err) : resolve(this.lastID) }
-        )
+        const values = Object.values(data)
+
+        if (!prep) {
```

Comment:
What's the purpose of prep?

Reply:
Prepared database queries. Initially was just playing around to speed things up when I was doing a lot of database initialisation and it was really slow. Not sure if it actually helps, but yeah.